src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -152,6 +155,25 @@ void GlobalAppSettings::LayerJson(const Json::Value& json, const OriginTag origi
     LayerActionsFrom(json, origin, true);
 
     JsonUtils::GetValueForKey(json, LegacyReloadEnvironmentVariablesKey, _legacyReloadEnvironmentVariables);
+    if (json[LegacyReloadEnvironmentVariablesKey.data()])
+    {
+        _logSettingSet(LegacyReloadEnvironmentVariablesKey);
+    }
+
+    // Remove settings included in userDefaults
+    static constexpr std::array<std::pair<std::string_view, std::string_view>, 2> userDefaultSettings{ { { "copyOnSelect", "false" },
+                                                                                                         { "copyFormatting", "false" } } };
+    for (const auto& [setting, val] : userDefaultSettings)
+    {
+        if (const auto settingJson{ json.find(&*setting.cbegin(), (&*setting.cbegin()) + setting.size()) })
+        {
+            if (settingJson->asString() == val)
+            {
+                // false positive!
+                _changeLog.erase(std::string{ setting });

```

Comment:
you shouldn't need to make a string out of this. If you do, look into "transparent containers" and see if there's something we can do about it.

Reply:
I still get a compiler error saying...
```
error C2665: 'std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::erase': no overloaded function could convert all the argument types
error C2665:         with
error C2665:         [
error C2665:             _Kty=std::string,
error C2665:             _Pr=std::less<std::string>,
error C2665:             _Alloc=std::allocator<Json::String>
error C2665:         ]
```

I tried declaring the set with a transparent comparator like this: `std::set<std::string, std::less<>>`
but still got basically the same error...
```
error C2665: 'std::_Tree<std::_Tset_traits<_Kty,_Pr,_Alloc,false>>::erase': no overloaded function could convert all the argument types
error C2665:         with
error C2665:         [
error C2665:             _Kty=std::string,
error C2665:             _Pr=std::less<void>,
error C2665:             _Alloc=std::allocator<Json::String>
error C2665:         ]
```

From the research I did, it looks like we'd have to build our own comparator to handle comparing strings and string_views, but we're only ctoring the string twice per JSON load. Is it worth adding in the extra comparator or should we just keep it as-is?